src/js/worklets/header_animations.js

```diff
@@ -0,0 +1,25 @@
+import {
+  useDerivedValue,
+  interpolate,
+  interpolateColor,
+  Extrapolation,
+  useAnimatedStyle,
+} from 'react-native-reanimated';
+
+const CLAMP_MIN = 0;
+const CLAMP_MAX = 60;
+const BLUR_MIN = 1;
+const BLUR_MAX = 15;
+
+export const useBlurAmount = (sharedValue) =>
+  useDerivedValue(() =>
+    parseInt(interpolate(sharedValue.value, [CLAMP_MIN, CLAMP_MAX], [BLUR_MIN, BLUR_MAX], Extrapolation.CLAMP)),
+  );
+
+export function useLayerOpacity(sharedValue, from, to) {
+  return useAnimatedStyle(() => ({
```

Comment:
This approach has few drawbacks.
1. I am not sure if situation is changed, but changes in js code were not visible until app restarted
2. Also for consistency we should stick to a single approach. If we use this approach everywhere we will have three files, view, simple style, and animated style, for all animations and we have several animations.

Reply:
> I am not sure if situation is changed, but changes in js code were not visible until app restarted

It hasn't changed :cry: and I know, it's a shame that we need to restart the app, I'm not sure if the REPL works to reload the JS requires.

> Also for consistency we should stick to a single approach. If we use this approach everywhere we will have three files, view, simple style, and animated style, for all animations and we have several animations.

Yes, I'm not a big fan of this approach too, that's why I first tried to solve it in a different way:
- https://github.com/status-im/status-mobile/pull/18795

although, I don't exactly like it, it solves the following existing problems:

1. We are always transforming CLJS styles to JS (which we know is expensive)
2. `apply-animations-to-style` is using a hook inside, but devs don't know it, and we can find A LOT of usages of this hook as a regular function, so it's not declared at the beginning, for example.
3. Our `interpolate-value` is not just a call to reanimated's interpolate (which is just a regular function), it hides a hook (`useDerivedValue`) so it has the same problems as mentioned in `2` Additionally,  we are making the code more complex, since instead of interpolate directly inside a `useAnimatedStyle`, we generate derived values with our `interpolate`. transform CLJS -> JS values, then pass them to `apply-animations-to-style` and transform them again.


To show the complexity added because of our wrappers, take a look at this piece of code (taken from the PR I mentioned, it is/was used to animate the avatar size).

#### With our wrappers 

```clojure
(defn f-avatar
  [...]
  ;; Interpolate, which is creating a derived value, it's a hook but looks as a regular function
  (let [image-scale-animation       (reanimated/interpolate scroll-y
                                                            scroll-animation-input-range
                                                            [1 0.4]
                                                            header-extrapolation-option)
        ;; 2nd interpolate usage, CLJS->JS transform, and useDerivedValue hook implied
        image-top-margin-animation  (reanimated/interpolate scroll-y
                                                            scroll-animation-input-range
                                                            [0 20]
                                                            header-extrapolation-option)
        ;; 3rd interpolate usage, CLJS->JS transform, and useDerivedValue hook implied
        image-side-margin-animation (reanimated/interpolate scroll-y
                                                            scroll-animation-input-range
                                                            [0 -20]
                                                            header-extrapolation-option)
        theme                       (quo.theme/get-theme)]
    [reanimated/view
     ;; `style/avatar-container` Looks just as a regular function, but it is/was using
     ;; `apply-animations-to-style`, which is terrible, because that's a hook, and we
     ;; aren't declaring it at the top, also it looks as a style fn, but it does more.
     {:style (style/avatar-container theme
                                     image-scale-animation
                                     image-top-margin-animation
                                     image-side-margin-animation)}
     ...]))
```

#### Without our wrappers

```clojure
(defn f-avatar
  [...]
  (let [theme            (quo.theme/get-theme)
        avatar-animation (worklets/use-avatar-animation scroll-y input-range #js[1 0.4] #js[0 20] #[0 -20])]
    [reanimated/view {:style [(style/avatar-container theme) ;; <- actually only styles
                              avatar-animation]}
     ...]))
```

```js
export function useAvatarAnimation(scrollY, inputRange, range1, range2, range3) {
  return useAnimatedStyles(() => (
      {
        transform: [
// Simplifyed calcs, maybe there were some other operations, but we can perform the interpolate here
// No need of extra sharedValues created with `useDerivedValue`
          {scale: interpolate(scrollY.value, inputRange.value, range1)},
          {translateX: interpolate(scrollY.value, inputRange.value, range2)},
          {translateX: interpolate(scrollY.value, inputRange.value, range3)}
        ]
      })
  )
}
```

The code is smaller and more straightforward, additionally, we are following the recommended practices for hooks.


@Parveshdhull WDYT?

Reply:
Thank you very much for sharing detailed explanation and example.

> apply-animations-to-style is using a hook inside, but devs don't know it, and we can find A LOT of usages of this hook as a regular function, so it's not declared at the beginning,

I think this problem can be easily solved. We just need to communicate to devs and add to guideline.

From your example, it seems for the use case of interpolate it looks simpler to just define whole style in js.
But for other use case, I think we should use our wrapper for now.
Otherwise we will lose the ability of hot reloading style related changes. And debugging and developing will be harder.

Reply:
We probably should rename this wrapper then. thank you @J-Son89

Reply:
@Parveshdhull we could:

1. Rename the wrappers adding `use-...` to the ones that involve a hook and keep using them.
2. Be aware that sometimes writing a worklet directly in JS may be cleaner and simpler in terms of code, so we can take the JS approach, for example, for complex animations (like the ones having multiple interpolations).

wdyt?

I've tried to create worklets in CLJS in different ways and I've got partially working results. I'm just trying to find a way to improve our animated code and API :upside_down_face:

Reply:
sounds good to me. Thank you for looking into this.